app/Modules/Media/tests/MediaTest.php

```diff
@@ -0,0 +1,320 @@
+<?php
+
+namespace App\Modules\Media\Tests;
+
+use App\Models\User;
+use App\Modules\Media\Models\Media;
+use App\Tests\Support\MediaTestTrait;
+use Illuminate\Http\Testing\File;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+use RonasIT\Support\Traits\FilesUploadTrait;
+
+class MediaTest extends ModuleTestCase
+{
+    use FilesUploadTrait;
+    use MediaTestTrait;
+
+    protected $admin;
+    protected $user;
+    protected File $file;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->admin = User::find(1);
+        $this->user = User::find(2);
+        $this->file = UploadedFile::fake()->image('file.png', 600, 600);
+
+        $this->mockGenerateFilename();
+    }
+
+    public function testCreate(): void
+    {
+        $response = $this->actingAs($this->admin)->json('post', '/media', ['file' => $this->file]);
+
+        $response->assertOk();
+
+        $responseData = $response->json();
+
+        $this->assertDatabaseHas('media', [
+            'id' => $responseData['id'],
+            'name' => 'file.png',
+            'owner_id' => $this->admin->id,
+            'is_public' => false,
+            'link' => '/storage/file.png'
+        ]);
+    }
+
+
+    public function testCreatePublic(): void
+    {
+        $response = $this->actingAs($this->user)->json('post', '/media', [
+                'file' => $this->file,
+                'is_public' => true,
+            ]
+        );
+
+        $responseData = $response->json();
+
+        $this->assertDatabaseHas('media', [
+            'id' => $responseData['id'],
+            'name' => 'file.png',
+            'owner_id' => $this->user->id,
+            'is_public' => true
+        ]);
+
+        $response->assertOk();
+    }
+
+    public function testCreateCheckUrls(): void
+    {
+        $this->actingAs($this->admin)->json('post', '/media', ['file' => $this->file]);
+
+        $this->assertEquals(1, Media::where('link', 'like', '/%')->count());
+    }
+
+    public function testCreateCheckResponse(): void
+    {
+        $response = $this->actingAs($this->admin)->json('post', '/media', ['file' => $this->file]);
+
+        $responseData = $response->json();
+
+        $this->assertDatabaseHas('media', [
+            'id' => $responseData['id'],
+            'link' => $responseData['link']
```

Comment:
let's mock link generation logic to check database changes without getting response data

Reply:
Done